tests/src/Unit/SetRolesTest.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace Drupal\Tests\og\Unit;
+
+use Drupal\og\Entity\OgMembership;
+use Drupal\og\Entity\OgRole;
+use Drupal\og\OgRoleInterface;
+use Drupal\Tests\UnitTestCase;
+
+/**
+ * Tests set role method from the og membership.
+ *
+ * @group og
+ * @coversDefaultClass \Drupal\og\Entity\OgMembership
+ */
+class SetRolesTest extends UnitTestCase {
```

Comment:
This should be in an `OgMembershipTest` class, as here we test only a specific method, but it can be extended